src/basic/log.c

```diff
@@ -1278,8 +1278,9 @@ int log_set_max_level_from_string(const char *e) {
         return 0;
 }
 
-int log_max_levels_to_string(int level, char **ret) {
+int log_max_levels_to_string(int clamp_level, char **ret) {
         _cleanup_free_ char *s = NULL;
+        int level = clamp_level >= 0 ? clamp_level : log_get_max_level();
```

Comment:
hmm, why this conditionalization here? seems unnecessary?

Or hmm, maybe i was confused before when I suggested to use LOG_NULL here as indicator. Maybe just introduce _LOG_LEVEL_MAX or so, as LOG_DEBUG + 1 (i.e. 8) and then pass that as clamp value to turn off clamping.

i always get confused with what's high and what's low with log levels... sorry for the confusion.

Reply:
hmm, so after reading on i guess this can stay that way, but maybe replace the rhs of the ternary op with a simple LOG_DEBUG. given that log_get_max_level() just returns the highest log level anyway, the `<= level` chec in the loop is redundant  in that case, and you achieve that with a simple LOG_DEBUG too

Reply:
@poettering It's not just about the check in the for loop though, we also serialize the log level itself just before the for loop. That's why I'm doing it like this.

Reply:
hmm, but shouldn't the max level be subject to the clamping too?

Reply:
Hmm right OK let me rework